examples/server/server.cpp

```diff
@@ -1083,7 +1083,7 @@ struct server_context {
             }
 
             // check if there is any token to predict
-            if (stop_pos == std::string::npos || (!slot.has_next_token && !is_stop_full && stop_pos > 0)) {
+            if (stop_pos == std::string::npos || is_stop_full || (!slot.has_next_token && !is_stop_full && stop_pos > 0)) {
```

Comment:
I think this `if` will always evaluate to `true`:

- if `stop_pos == std::string::npos` -> `true`
- if `stop_pos != std::string::npos`, then `is_stop_full == true` due to line 1075 -> `true`

Reply:
I have attempted to simplify this logic in the new commit. The text will not be sent only when the partial match is found. And partial match will be searched for only when the current token is not the last one.